crypto/jwk.go

```diff
@@ -50,3 +75,238 @@ func ToPublicKeyJWK(key jwk.Key) (*PublicKeyJWK, error) {
 	}
 	return &pubKeyJWK, nil
 }
+
+// PublicKeyToPublicKeyJWK converts a public key to a PublicKeyJWK
+func PublicKeyToPublicKeyJWK(key crypto.PublicKey) (*PublicKeyJWK, error) {
+	switch key.(type) {
+	case rsa.PublicKey:
+		return JWKFromRSAPublicKey(key.(rsa.PublicKey))
+	case ed25519.PublicKey:
+		return JWKFromEd25519PublicKey(key.(ed25519.PublicKey))
+	case x25519.PublicKey:
+		return JWKFromX25519PublicKey(key.(x25519.PublicKey))
+	case secp256k1.PublicKey:
+		return JWKFromSECP256k1PublicKey(key.(secp256k1.PublicKey))
+	case ecdsa.PublicKey:
+		return JWKFromECDSAPublicKey(key.(ecdsa.PublicKey))
+	default:
+		return nil, fmt.Errorf("unsupported public key type: %T", key)
+	}
+}
+
+// PrivateKeyToPrivateKeyJWK converts a private key to a PrivateKeyJWK
+func PrivateKeyToPrivateKeyJWK(key crypto.PrivateKey) (*PublicKeyJWK, *PrivateKeyJWK, error) {
```

Comment:
nit:
Have you thought about making the JWKFromRSAPrivateKey functions private? It will be a lot less functions for a user to deal with and just have the global function available to them

Reply:
yeah, let me change this. can always make them public later if needed